modules/mobile/app/models/mobile/v0/adapters/vaos_v2_appointments.rb

```diff
@@ -267,33 +301,30 @@ def location(type, appointment_hash)
             end
             location[:lat] = appointment_hash.dig(:location, :lat)
             location[:long] = appointment_hash.dig(:location, :long)
-            location[:phone] = location_phone(appointment_hash)
+            location[:phone] = parse_phone(appointment_hash.dig(:location, :phone, :main))
           end
 
           location
         end
         # rubocop:enable Metrics/MethodLength
 
-        def location_phone(appointment_hash)
-          phone = appointment_hash.dig(:location, :phone, :main)
-
+        def parse_phone(phone)
           # captures area code (\d{3}) number (\d{3}-\d{4})
           # and optional extension (until the end of the string) (?:\sx(\d*))?$
-          phone_captures = phone&.match(/^(\d{3})-(\d{3}-\d{4})(?:\sx(\d*))?$/)
+          phone_captures = phone&.match(/^\(?(\d{3})\)?.?(\d{3})-?(\d{4})(?:\sx(\d*))?$/)
 
           if phone_captures.nil?
             Rails.logger.warn(
-              'mobile appointments failed to parse VAOS V2 facility phone number',
-              facility_id: appointment_hash.dig(:location, :id),
-              facility_phone: phone
+              'mobile appointments failed to parse VAOS V2 phone number',
+              phone: phone
             )
             return { area_code: nil, number: nil, extension: nil }
           end
 
           {
             area_code: phone_captures[1].presence,
-            number: phone_captures[2].presence,
-            extension: phone_captures[3].presence
+            number: "#{phone_captures[2].presence}-#{phone_captures[3].presence}",
```

Comment:
I do actually like this method better. Splicing it into the array was hard to read.